src/extension/tab/hook.ts

```diff
@@ -256,8 +262,12 @@ function initializeHook() {
     sendHookDataToDevTools("connectToDevtools");
   }
 
+  function handleClientMessage(message: ClientDevtoolsMessage) {
```

Comment:
This on its own seems to be working well, though per [my comment](https://github.com/apollographql/apollo-client/pull/11647/files#r1515609697) we would likely benefit from a message queue where we could replay messages here on initialization.